### ISSUE TITLE
HA update 2022.4.0 - all entities gone

### ISSUE THREAD

**Issue by xwox** [laszlojakab/homeassistant-easycontrols#66]
I just upgraded to 2022.4.0
All Helios/EasyControl entities are gone.

I get the following error messages:
--><---------------
Error adding entities for domain binary_sensor with platform easycontrols
11:56:51 – (FEHLER) Binärsensor - Die Nachricht ist zum ersten Mal um 11:56:51 aufgetreten und erscheint 2 mal
Error adding entities for domain sensor with platform easycontrols
11:56:51 – (FEHLER) Sensor - Die Nachricht ist zum ersten Mal um 11:56:51 aufgetreten und erscheint 2 mal
--><---------------

The following detailed error messages are also provided:
--><---------------
2022-04-07 11:56:51 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform easycontrols
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in _async_add_entity
entry = entity_registry.async_get_or_create(
File "/usr/src/homeassistant/homeassistant/helpers/entity_registry.py", line 383, in async_get_or_create
raise ValueError("entity_category must be a valid EntityCategory instance")
ValueError: entity_category must be a valid EntityCategory instance
2022-04-07 11:56:51 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entities for domain binary_sensor with platform easycontrols
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in _async_add_entity
entry = entity_registry.async_get_or_create(
File "/usr/src/homeassistant/homeassistant/helpers/entity_registry.py", line 383, in async_get_or_create
raise ValueError("entity_category must be a valid EntityCategory instance")
ValueError: entity_category must be a valid EntityCategory instance
2022-04-07 11:56:51 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up easycontrols platform for sensor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 257, in _async_setup_platform
await asyncio.gather(*pending)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in _async_add_entity
entry = entity_registry.async_get_or_create(
File "/usr/src/homeassistant/homeassistant/helpers/entity_registry.py", line 383, in async_get_or_create
raise ValueError("entity_category must be a valid EntityCategory instance")
ValueError: entity_category must be a valid EntityCategory instance
2022-04-07 11:56:51 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up easycontrols platform for binary_sensor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 257, in _async_setup_platform
await asyncio.gather(*pending)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
await asyncio.gather(*tasks)
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in _async_add_entity
entry = entity_registry.async_get_or_create(
File "/usr/src/homeassistant/homeassistant/helpers/entity_registry.py", line 383, in async_get_or_create
raise ValueError("entity_category must be a valid EntityCategory instance")
ValueError: entity_category must be a valid EntityCategory instance
--><---------------

**Comment by PhilippeRaven** [comment]
I just updated to 2022.4.0,  got exactly the same problem.

**Comment by MrSimpa** [comment]
The same issue for me, just fan.helios_kwl switch is there... I read that 2022.4.0 have problem with fan: ERROR - setup failed for custom integration : unable to import component: cannot import name 'speed_high' from 'homeassistant.components.fan' But i dont know how to fix it...

**Comment by xwox** [comment]
No changes to this problem after update to 2022.4.1

**Comment by scochet11** [comment]
same issue, same errors. Seems like something changed in HA and an entity configType must be set now. Other integrations had the same problems, for example: https://github.com/thomasgermain/vaillant-component

**Comment by djm193** [comment]
Got the same error in 2022.4.1 and tried to set it up new. After removing and adding it back it was faulty on the setup.
So not only entities are gone-the whole integration doesn’t work anymore.

Error setting up entry Helios None (Helios500) for easycontrols
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 335, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/easycontrols/__init__.py", line 57, in async_setup_entry
    await controller.init()
  File "/config/custom_components/easycontrols/controller.py", line 57, in init
    self._maximum_air_flow = float(re.findall(r'\d+', self._model)[0])
  File "/usr/local/lib/python3.9/re.py", line 241, in findall
    return _compile(pattern, flags).findall(string)
TypeError: expected string or bytes-like object

**Comment by PhilippeRaven** [comment]
@laszlojakab Do you think you could fix it in the same way as @scochet11 showed it in the "vaillant-component" repo ?
Thanks !

**Comment by laszlojakab** [comment]
I have just released 0.5.6-beta, please try this version and give a feedback!

**Comment by PhilippeRaven** [comment]
I just installed it. 
Everything is working again, "but" I got a "Unknown" value for "VMC errors" and "VMC warnings".
I _think_ it's normal with the new way of handling the "Unknown" state, but maybe there's a way to display "None" instead ?

**Comment by MrSimpa** [comment]
Beta version works for me, except Warnings, Errors and Information. Tahanks.

I have a question about party mode and fan boost, only for that reason i use other integration https://github.com/asev/homeassistant-helios too , can i do somethingh liske this in this easycontrols integration? I use it for specific time when i want fan boost and on another side when my windows are open for low fan mode, because i dont want to off it.

**Comment by scochet11** [comment]
Besides Errors and Information the integration is working again. The following Warning is logged: 

Logger: homeassistant.helpers.entity
Source: helpers/entity.py:745
First occurred: 11. April 2022, 22:11:54 (286 occurrences)
Last logged: 07:34:29

Update of sensor.helios_ec300_airflow_rate is taking over 10 seconds
Update of sensor.helios_ec300_heat_recovery_efficiency is taking over 10 seconds
Update of binary_sensor.helios_ec300_bypass is taking over 10 seconds
Update of binary_sensor.helios_ec300_filter_change is taking over 10 seconds
Update of fan.helios_ec300 is taking over 10 seconds

**Comment by xwox** [comment]
Thx a lot. Great work & fast too!

Version 0.5.6-beta works again for me, except for the following messages:

Logger: homeassistant.helpers.entity
Source: helpers/entity.py:745
First occurred: 08:12:40 (19 occurrences)
Last logged: 08:12:40

Update of sensor.kwl_airflow_rate is taking over 10 seconds
Update of sensor.kwl_heat_recovery_efficiency is taking over 10 seconds
Update of binary_sensor.kwl_bypass is taking over 10 seconds
Update of binary_sensor.kwl_filter_change is taking over 10 seconds
Update of fan.kwl is taking over 10 seconds

**Comment by laszlojakab** [comment]
New version 0.5.6-beta2 released where error, warning, information sensors are fixed too. The `taking over 10 seconds` warnings is an older problem. The querying of sensor values can't be done in parallel (it's the limitation of the protocol used by the Helios devices) so sensors have to wait until other sensor values are received.

**Comment by MrSimpa** [comment]
Thanks for quick response. But after restart server i get error: 

Logger: homeassistant.config_entries
Source: custom_components/easycontrols/controller.py:123
Integration: Helios Easy Controls ([documentation](https://github.com/laszlojakab/homeassistant-easycontrols#usage), [issues](https://github.com/laszlojakab/homeassistant-easycontrols/issues))
First occurred: 12:36:18 (1 occurrences)
Last logged: 12:36:18

Error setting up entry Helios KWL EC 300W ET R (Helios KWL) for easycontrols
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/streams.py", line 52, in open_connection
    transport, _ = await loop.create_connection(
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1041, in create_connection
    sock = await self._connect_sock(
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 955, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 502, in sock_connect
    return await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 492, in wait_for
    fut.result()
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 335, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/easycontrols/__init__.py", line 57, in async_setup_entry
    await controller.init()
  File "/config/custom_components/easycontrols/controller.py", line 54, in init
    self._model = await self.get_variable(VARIABLE_ARTICLE_DESCRIPTION)
  File "/config/custom_components/easycontrols/controller.py", line 123, in get_variable
    value = await self._eazyctrl.get_variable(
  File "/usr/local/lib/python3.9/site-packages/eazyctrl.py", line 481, in get_variable
    async with AsyncEazyCommunicator(self._server, *self._serverargs) as comm:
  File "/usr/local/lib/python3.9/site-packages/eazyctrl.py", line 327, in __aenter__
    self._reader, self._writer = await asyncio.wait_for(asyncio.open_connection(self._server, self._port), self._timeout)
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 494, in wait_for
    raise exceptions.TimeoutError() from exc
asyncio.exceptions.TimeoutError

If i readd integration it is OK, but after restart HA shows again.

**Comment by laszlojakab** [comment]
@MrSimpa it looks like a timeout error during initialization. I think it is a different issue, maybe the integration should throw a PlatformNotReady error (https://developers.home-assistant.io/docs/integration_quality_scale_index#silver-) for that case. I'll open a new issue for that.

**Comment by laszlojakab** [comment]
Fix released in version 0.5.6, closing the issue.

**Comment by djm193** [comment]
Thanks a lot! I have managed to install it again but only the switch remains „not broken“.
Cheers 
Daniel

Von meinem iPhone gesendet

**Comment by djm193** [comment]
Works for me! Thanks m!

Von meinem iPhone gesendet
